rfcs/20220221-3d_model_support.md

```diff
@@ -0,0 +1,181 @@
+| `updated_on` | `timestamp` | Time the record was updated |
+| `ingestion_type` | `varchar` | What generated the record |
+| `provider` | `varchar` | Provider API that the model comes from |
+| `source` | `varchar` | Source within the provider (e.g. NASA on Flickr), typically the same as `provider` |
+| `foreign_id` | `varchar` | Identifier that the provider uses for this model |
+| `foreign_landing_url` | `varchar` | URL on the provider that displays the "single-result" page for this model |
+| `url` | `varchar` | URL to the actual asset itself (e.g. the image URL) |
+| `thumbnail` | `varchar` | Thumbnail URL |
+| `filesize` | `integer` | Size of the file |
+| `license` | `varchar` | License for the model |
+| `license_version` | `varchar` | Version of the model's license |
+| `creator` | `varchar` | Creator ID of the model from the provider |
+| `creator_url` | `varchar` | URL to the creator's page on the provider's site |
+| `title` | `varchar` | Title of the model |
+| `meta_data` | `jsonb` | Uncategorized metadata for the model |
+| `tags` | `jsonb` | Tags that describe the model, for use in searching |
+| `watermarked` | `bool` | Whether the model is watermarked - is this needed for 3D models? |
+| `last_synced_with_source` | `timestamp` | Time the record was last updated from the provider |
+| `removed_from_source` | `bool` | Whether the model has been removed from the upstream source |
+| `category` | `varchar` | Category for the model (is this where we'll make the distinction between 3D printers, game assets, etc.?) |
+| `filetype` | `varchar` | File type of the primary file for the model |
+| `alt_files` | `jsonb` | Additional file types for the model |
+| `vertices` | `integer` | Number of vertices within the model |
+| `textures` | `integer` | Number of [textures](https://en.wikibooks.org/wiki/Blender_3D:_Noob_to_Pro/Materials_and_Textures) the model uses |
+| `materials` | `integer` | Number of [materials](https://en.wikibooks.org/wiki/Blender_3D:_Noob_to_Pro/Materials_and_Textures) the model uses |
+| `uv_mapping` | `bool` | Whether the model contains [UV Mapping](https://en.wikipedia.org/wiki/UV_mapping) |
+| `vertex_colors` | `bool` | Whether the model contains [vertex colors](https://help.sketchfab.com/hc/en-us/articles/210765623-Vertex-Colors) |
+| `pbr` | `bool` | Whether the model supports [physically based rendering](https://en.wikipedia.org/wiki/Physically_based_rendering) |
+| `rigged_geometries` | `bool` | Whether the model supports [rigging for animation](https://dreamfarmstudios.com/blog/what-is-3d-rigging/) |
+| `animations` | `integer` | Number of animations the model has |
+| `morph_geometries` | `integer` | Number of [morph targets for animation](https://en.wikipedia.org/wiki/Morph_target_animation) |
+
+
+# 3D Models in the frontend
+
+## Some potential risks
+
+- 3D models are often heavy files. We should lazy-load and defer loading of these resources as much as possible.
```

Comment:
What do we need to do for this? Does this just mean using intersection observer or something like that to prevent them from loading?

If IntersectionObserver isn't available on a platform, should we fall back to deferring loading some other way?

Do we need to try to eagerly fetch resources under certain circumstances (maybe we start loading the 3d model after hover or focusing on a result for a certain amount of time).

We probably need to invest more time into figuring out how we're going to demonstrate that the resource is still loading. We have [this issue for adding a loading animation for the audio](https://github.com/WordPress/openverse-frontend/issues/1001) (in fact I thought that issue already existed but it must have fell through the cracks when we discussed it). Whatever we do there could inform what we do for 3d models as well.